src/reward_splitter/tasks.py

```diff
@@ -0,0 +1,246 @@
+
+            tx_receipt = await execution_client.eth.wait_for_transaction_receipt(
+                tx, timeout=settings.execution_transaction_timeout
+            )
+            if not tx_receipt['status']:
+                raise RuntimeError(
+                    f'Failed to confirm reward splitter tx: {tx_hash}',
+                )
+            logger.info('Transaction %s confirmed', tx_hash)
+
+        app_state.reward_splitter_block = block['number']
+        logger.info('All multicall batches processed')
+
+
+async def _check_reward_splitter_block(app_state: AppState, block_number: BlockNumber) -> bool:
+    last_processed_block = app_state.reward_splitter_block
+    reward_splitter_blocks_interval = (
+        REWARD_SPLITTER_INTERVAL // settings.network_config.SECONDS_PER_BLOCK
+    )
+    if (
+        last_processed_block
+        and last_processed_block + reward_splitter_blocks_interval >= block_number
+    ):
+        return False
+    return True
+
+
+async def _get_reward_splitter_calls(
+    reward_splitter: RewardSplitter,
+    harvest_params: HarvestParams | None,
+    exit_requests: list[ExitRequest],
+) -> list[HexStr]:
+    """
+    Generate ABI encoded calls for the reward splitter contract.
+
+    It includes calls:
+    1. for updating the vault state,
+    2. entering the exit queue on behalf of shareholders,
+    3. claiming exited assets on behalf of shareholders.
+
+    Returns:
+        list[HexStr]: A list of ABI encoded calls for the reward splitter contract.
+    """
+    # ABI encoded calls without contract address
+    reward_splitter_calls: list[HexStr] = []
+    reward_splitter_encoder = _get_reward_splitter_encoder()
+
+    reward_splitter_assets = await _get_reward_splitter_assets(reward_splitter)
+
+    if reward_splitter_assets < REWARD_SPLITTER_MIN_ASSETS:
+        logger.info('Reward splitter %s has not enough assets to withdraw', reward_splitter.address)
+    else:
+        # Append update state call
+        if harvest_params:
+            reward_splitter_calls.append(
+                reward_splitter_encoder.update_vault_state(harvest_params=harvest_params)
+            )
+
+        # Append enter exit queue on behalf calls
+        for shareholder in reward_splitter.shareholders:
+            logger.info('Processing shareholder %s', shareholder.address)
+
+            reward_splitter_calls.append(
+                reward_splitter_encoder.enter_exit_queue_on_behalf(
+                    rewards=None,  # exiting all rewards
+                    address=shareholder.address,
+                )
+            )
+
+    # Append claim exited assets on behalf calls
+    if not exit_requests:
+        logger.info('No exit requests for reward splitter %s', reward_splitter.address)
+
+    for exit_request in exit_requests:
+        logger.info('Processing exit request with position ticket %s', exit_request.position_ticket)
+        if exit_request.exit_queue_index is None:
+            logger.info(
```

Comment:
claimable exit request with no queue index is unexpected behaviour
```suggestion
            logger.error(
```